lib/filesystem/include/filesystem/filesystem.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef IRODS_FILESYSTEM_FILESYSTEM_HPP
+#define IRODS_FILESYSTEM_FILESYSTEM_HPP
+
+#include "filesystem/config.hpp"
+#include "filesystem/object_status.hpp"
+#include "filesystem/permissions.hpp"
+#include "filesystem/copy_options.hpp"
+
+#include "rcConnect.h"
+
+#include <cstdint>
+#include <string>
+#include <istream>
+#include <ostream>
+#include <chrono>
+#include <vector>
+
+#include <boost/variant.hpp>
+
+namespace irods {
+namespace experimental {
+namespace filesystem {
+
+    class path;
+
+    using object_time_type = std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds>;
+
+    enum class remove_options
+    {
+        none,
+        no_trash
+    };
+
+    struct checksum
+    {
+        int replica_number;
+        std::string value;
+        std::uintmax_t size;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [struct member 'checksum::size' is never used.](https://app.codacy.com/app/trel/irods/pullRequest?prid=4593032)